packages/components/src/dropdown/index.tsx

```diff
@@ -1,21 +1,26 @@
-// @ts-nocheck
 /**
  * External dependencies
  */
 import classnames from 'classnames';
+import type { ForwardedRef } from 'react';
 
 /**
  * WordPress dependencies
  */
-import { useRef, useEffect, useState } from '@wordpress/element';
+import { forwardRef, useEffect, useRef, useState } from '@wordpress/element';
 import { useMergeRefs } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import Popover from '../popover';
+import type { WordPressComponentProps } from '../ui/context';
+import type { DropdownProps } from './types';
 
-function useObservableState( initialState, onStateChange ) {
+function useObservableState(
+	initialState: boolean,
+	onStateChange?: ( newState: boolean ) => void
+): [ boolean, ( newState: boolean ) => void ] {
```

Comment:
You'd hope that TS would infer the return type from the `return` statement below, but it doesn't.

Reply:
Whoa, thanks! So much cleaner to let TS infer the type.

Committed in https://github.com/WordPress/gutenberg/pull/45787/commits/1f9ff853fa72a2a784be08449b7f99561bc647ad